addons/hr_expense/models/hr_expense.py

```diff
@@ -151,13 +151,21 @@ def _compute_state(self):
             else:
                 expense.state = "done"
 
-    @api.depends('quantity', 'unit_amount', 'tax_ids', 'currency_id')
+    @api.depends('total_amount', 'quantity', 'unit_amount', 'tax_ids', 'currency_id')
```

Comment:
You depend on a field that is computed in that same method :thinking: 
I get that it's for the taxes, but I guess the compute should be split then

Reply:
Hi Kevin, thanks for checking! Actually, I added this dependency to tackle the case when a user duplicates an existing expense and then changes the product. For example,
1. create an expense of a product with unit price, make the sub_total field always visible in the form view
2. duplicate it, and change the product to one without unit price (where we enter the total_amount)
3. enter the total_amount, but if without this dependency, the computation isn't triggered and thus incorrect

Does this explanation make sense to you? Cheers! :smile:

Reply:
The explanation makes sense, but I'm afraid that's not the correct way to do it tho :-( 
It's introducing a circular dependency, which is why @sofiagvaladze had to split the compute in 15.2[1].

I believe you'll have to do the same thing here

[1] : https://github.com/odoo/odoo/blob/master/addons/hr_expense/models/hr_expense.py#L173